core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -247,7 +251,26 @@ private void doExecute0(ClientRequestContext ctx, HttpRequestDuplicator rootReqD
         final HttpResponse response = executeWithFallback(delegate(), derivedCtx,
                                                           (context, cause) -> HttpResponse.ofFailure(cause));
 
-        derivedCtx.log().whenAvailable(RequestLogProperty.RESPONSE_HEADERS).thenAccept(log -> {
+        if (requiresResponseTrailers) {
+            response.aggregate().handle((aggregated, cause) -> {
+                handleResponse(ctx, rootReqDuplicator, originalReq, returnedRes, future, derivedCtx,
+                               cause != null ? response : aggregated.toHttpResponse());
```

Comment:
I think we should pass `HttpResponse.ofFailure(cause)` instead of `response`.
If the rule does not retry on the exception, then the `response` will be scribed twice.

Reply:
Nice catch! Fixed.